### ISSUE TITLE
KeyError: 'DOCUTILSCONFIG' when running `make html`

### ISSUE THREAD

**Issue by Dmitrii-I** [sphinx-doc/sphinx#5928]
Subject: <what happen when you do on which document project>

### Problem
I had Sphinx 1.6.*. Documentation was building fine. I upgraded Sphinx to 1.8.3. After that, `make html` was giving an error:

```
/usr/local/bin/sphinx-build:11: RemovedInSphinx20Warning: `sphinx.main()` has moved to `sphinx.cmd.build.main()`.
  sys.exit(main())
Running Sphinx v1.8.3

Exception occurred:
  File "/usr/lib/python3.5/os.py", line 725, in __getitem__
    raise KeyError(key) from None
KeyError: 'DOCUTILSCONFIG'
```

#### Procedure to reproduce the problem
```
make html
```

#### Error logs / results
```
# Sphinx version: 1.8.3
# Python version: 3.5.2 (CPython)
# Docutils version: 0.14 
# Jinja2 version: 2.8
# Last messages:

# Loaded extensions:
Traceback (most recent call last):
  File "/home/dima/.local/lib/python3.5/site-packages/sphinx/cmd/build.py", line 305, in build_main
    return app.statuscode
  File "/usr/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/dima/.local/lib/python3.5/site-packages/sphinx/util/docutils.py", line 144, in patch_docutils
    yield
  File "/usr/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/dima/.local/lib/python3.5/site-packages/sphinx/util/docutils.py", line 134, in using_user_docutils_conf
    os.environ.pop('DOCUTILSCONFIG')
  File "/usr/lib/python3.5/_collections_abc.py", line 727, in pop
    value = self[key]
  File "/usr/lib/python3.5/os.py", line 725, in __getitem__
    raise KeyError(key) from None
KeyError: 'DOCUTILSCONFIG'
```

#### Expected results
I want to build html documentation.

### Reproducible project / your project
https://github.com/Dmitrii-I/sportmonks

### Environment info
- OS: Ubuntu 16.04.5 LTS 32-bit
- Python version: 3.5.2 
- Sphinx version: 1.8.3

**Comment by Dmitrii-I** [comment]
It seems that the issue is in my Makefile where my source dir was the current dir, which lead to the above described bug. Here is the Makefile causing the issue:

```
# Minimal makefile for Sphinx documentation
#

# You can set these variables from the command line.
SPHINXOPTS    =
SPHINXBUILD   = sphinx-build
SPHINXPROJ    = sportmonks
SOURCEDIR     = 
BUILDDIR      = build

# Put it first so that "make" without argument is like "make help".
help:
	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)

.PHONY: help Makefile

# Catch-all target: route all unknown targets to Sphinx using the new
# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
%: Makefile
	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)

```

As a workaround I changed `SOURCEDIR     = ` to `SOURCEDIR     = source`, moved the *.rst files there and ran `make html` and it worked.

**Comment by tk0miya** [comment]
Fixed by #5931 .
Thank you for reporting!
